src/plugins/validation/2and3/semantic-validators/parameters-ibm.js

```diff
@@ -182,3 +192,17 @@ function formatValid(obj, isOAS3) {
   }
   return false;
 }
+
+function checkInlineParameters(obj, isRef, parameterNames, messages, path) {
+  if (!isRef) {
+    if (!parameterNames.includes(obj.name)) {
```

Comment:
I don't think that only checking the name is sufficient for recommending a referenced parameter. If these parameters have the same name but are different (which is maybe a different issue), they cannot be consolidated into a referenced parameter. This check should assert a stricter level of equality before determining there is an issue.